[16.0][MIG]stock_location_orderpoint : Migration to 16.0

mt-software-de opened the pull request:
This is the migration of https://github.com/OCA/stock-logistics-warehouse/pull/1636

cc @rousseldenis @jbaudoux @lmignon

mt-software-de committed:
[ADD] stock_location_orderpoint:

Declare orderpoint on a location allowing to replenish any product with the same criteria

rousseldenis committed:
[IMP] stock_location_orderpoint: Add access to orderpoints from location form

rousseldenis committed:
[IMP] stock_location_orderpoint: Add demo data to ease tests

rousseldenis committed:
[IMP] stock_location_orderpoint: Add CONFIGURE readme fragment

rousseldenis committed:
[IMP] stock_location_orderpoint: Improve loop and add a test

Commit:
[UPD] Update stock_location_orderpoint.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
stock_location_orderpoint 14.0.1.0.1

mt-software-de committed:
[IMP] stock_location_orderpoint: black, isort, prettier

mt-software-de committed:
[MIG] stock_location_orderpoint: Migration to 16.0

LoisRForgeFlow commented on stock_location_orderpoint/README.rst:
```diff
@@ -0,0 +1,140 @@
+Location Orderpoint configuration
+=================================
+
+#. Go either by the stock location you want to replenish and click on 'Orderpoints'
+   or go to Inventory > Configuration > Settings > Warehouse > Stock Location Orderpoint
```
The menu route here is incorrect:

![image](https://github.com/OCA/stock-logistics-orderpoint/assets/23449160/f62049be-8c0b-467d-8899-8ff7d1c39e3a)

Also, you can see that there is a problem in the form view with the active toggle button

mt-software-de replied:
thanks a lot i will fix this

mt-software-de replied:
![image](https://github.com/OCA/stock-logistics-orderpoint/assets/78783475/238b2c23-abed-4931-85b5-c3d597206d3a)
![image](https://github.com/OCA/stock-logistics-orderpoint/assets/78783475/7e6df109-356c-4aff-86bd-651b2d6f66dd)
![image](https://github.com/OCA/stock-logistics-orderpoint/assets/78783475/8fd5cc31-e2c0-4595-95ac-ea4c44898457)

I added a ribbon

LoisRForgeFlow reviewed (commented):
Interesting module

mt-software-de committed:
[FIX] stock_location_orderpoint: Fix menu location and views

mt-software-de committed:
[IMP] stock_location_orderpoint: Add an index on stock move level

If the database contains billion of stock moves, deletion of an orderpoint can
take a great amount of time. Add an index to avoid that.

mt-software-de commented:
@rousseldenis 
Could you maybe approve this one now? and merge it?

rousseldenis commented:
> @rousseldenis Could you maybe approve this one now? and merge it?

Nearly, we are in final phase of tests on our side.

jbaudoux commented on stock_location_orderpoint/models/stock_location_orderpoint.py:
```diff
@@ -0,0 +1,413 @@
+    _sql_constraints = [
+        (
+            "location_route_unique",
+            "unique(location_id, route_id)",
```
We need one rule per replenish_method
```suggestion
            "unique(location_id, route_id, company_id, replenish_method)",
```

jbaudoux commented on stock_location_orderpoint/views/stock_location_orderpoint_views.xml:
```diff
@@ -0,0 +1,85 @@
+                <field name="trigger" />
+                <field name="replenish_method" />
+                <field name="group_id" />
+                <field
```
```suggestion
                <field name="priority" optional="hide" />
                <field
```

mt-software-de committed:
[IMP] stock_location_orderpoint: sql unique constraint, tree view
- Improve sql unique constraint
- Add priority field to tree view

jbaudoux commented on stock_location_orderpoint/models/stock_move.py:
```diff
@@ -0,0 +1,106 @@
+        job_options = job_options.copy()
+        job_options.setdefault(
+            "description",
+            _("Try to replenish quantities for location {} and product {}").format(
```
@mt-software-de Not related to the migration but if the location_field is `location_src_id` then it would be more understandable if the job was named "Try to replenish quantities **from** location".

It's a detail but when you try to understand what's going on, this is helpful

mt-software-de replied:
please have a look now.

mt-software-de committed:
[IMP] stock_location_orderpoint: Improve queue.job description

rousseldenis commented:
@mt-software-de Could you just add a little description on first commit line and we can merge ?

![image](https://github.com/OCA/stock-logistics-orderpoint/assets/19529533/8450a04d-74f0-4e56-a487-d4d5dc337392)

mt-software-de commented:
> @mt-software-de Could you just add a little description on first commit line and we can merge ?
> 
> ![image](https://user-images.githubusercontent.com/19529533/270902343-8450a04d-74f0-4e56-a487-d4d5dc337392.png)

done

mt-software-de commented:
> > @mt-software-de Could you just add a little description on first commit line and we can merge ?
> > ![image](https://user-images.githubusercontent.com/19529533/270902343-8450a04d-74f0-4e56-a487-d4d5dc337392.png)
> 
> done

ping @rousseldenis could we merge this one now?

rousseldenis commented:
/ocabot merge nobump

rousseldenis commented:
@mt-software-de Could you fix pre-commit? Some rules have been enforced.

mt-software-de commented:
> @mt-software-de Could you fix pre-commit? Some rules have been enforced.

done

rousseldenis commented:
/ocabot merge nobump
